### PR TITLE
renameSystemCharts + fix in upsert

### PR DESCRIPTION
**Pull request by leonPep42**

### CODE REVIEW

File: server-side/chart-service.ts [hunk 1]
```diff
@@ -27,7 +27,9 @@ class ChartService {
         const body = request.body;
 
         //system charts keys will contain the addon uuid suffix
-        body.Key = body.System ? `${body.Name}_c2cc9af4d7ad.js` : `${body.Name}.js`; 
+        if(body.Hidden == false)
```
Comment (ShirMarciano) [inline server-side/chart-service.ts:30]: if hidden= true we dont need to set the key? 
It will be undefined?

File: server-side/chart-service.ts [hunk 2]
```diff
@@ -71,7 +73,7 @@ class ChartService {
                 file.Hidden = true;
             }
         
-            return await this.papiClient.post(`/addons/pfs/${this.client.AddonUUID}/${CHARTS_PFS_TABLE_NAME}`,file);
+            return this.papiClient.post(`/addons/pfs/${this.client.AddonUUID}/${CHARTS_PFS_TABLE_NAME}`,file);
```
Comment (ShirMarciano) [inline server-side/chart-service.ts:76]: why do you remove the await?

File: server-side/installation.ts [hunk 3]
```diff
@@ -62,4 +63,26 @@ async function createDIMXRelations(client: Client) {
...
+}
+
+export async function renameSystemCharts(client: Client) {
+    const service = new ChartService(client)
+    let systemCharts = await service.find({where: 'System=true'});
+    if(systemCharts[0].Key.includes("c2cc9af4d7ad")) {
+        console.log("system charts already updated")
+    }
+    else {
+        for(const chartIndex in systemCharts) {
```
Comment (ShirMarciano) [inline server-side/installation.ts:75]: where is the part you put the suffix?

File: server-side/installation.ts [hunk 4]
```diff
@@ -62,4 +63,26 @@ async function createDIMXRelations(client: Client) {
...
+            //deleting old chart, upsert will use the old key because hidden=true
+            console.log(systemCharts[chartIndex])
+            systemCharts[chartIndex].Hidden = true;
+            await service.papiClient.post('/charts',systemCharts[chartIndex]);
+
+            // now it will be saved with the new key, because hidden=false
+            systemCharts[chartIndex].Hidden = false;
```
Comment (ShirMarciano) [inline server-side/installation.ts:82]: where is the part you put the suffix?

File: server-side/chart-service.ts (hunk 1 above)
Comment (leonPep42) [reply server-side/chart-service.ts:30]: if hidden=true it means key already exists
before I added the if, trying to hide old system chart would change the key to the new format and then the old chart with the old key is not hided

**Pull request merged**